### PR TITLE
Size mapping functionality

### PR DESCRIPTION
**Pull request by mkendall07**
## Type of change

- [x] Feature

## Description of change
Adding support for https://github.com/prebid/Prebid.js/issues/87 - size mapping. 

- Usage:
```
var adUnit = {
  code: 'code',
  sizeMapping: [  //new!
    {
      minWidth : 1024,  //if device screen width is greater than 1024, use these sizes
      sizes : [[300,250],[728,90]]
    },
    {
      minWidth : 480,  //if device screen width is < 1024 && > 480, use these sizes
      sizes : [120,60]
    },
    {
      minWidth : 0,  //if device screen width is < 480 && > 0, use these sizes
      sizes : [20,20]
    }
  ]
  bids: [...]
}
```

## Other information

@prebid/core  for review.

### CODE REVIEW

File: src/sizeMapping.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * @module sizeMapping
+ */
+import * as utils from './utils';
+
+exports.mapSizes = function(adUnit) {
+  if(!isSizeMappingValid(adUnit.sizeMapping)){
+    return adUnit.sizes;
+  }
+  const width = this.getScreenWidth();
```
Comment (matthewlane) [inline src/sizeMapping.js]: On a test page, this line gave me `TypeError: Cannot read property 'getScreenWidth' of undefined` due to this module using CommonJS-style exports but adapter manager using ES2015-style imports. Line 4 of adaptermanager could be changed to use a require statement, or the functions in this module could be exported with the ES2015 `export` keyword--in that case the unit tests fail though and would need to be changed as well

File: src/sizeMapping.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${adUnit.sizes}`);
+  return sizes;
+
+};
+
+function isSizeMappingValid(sizeMapping) {
+  if(utils.isArray(sizeMapping) && sizeMapping.length > 0){
+    return true;
+  }
+  utils.logError('sizeMapping needs at least one screen size defined');
```
Comment (matthewlane) [inline src/sizeMapping.js]: On pages that haven't set up any `sizeMapping`s but are still using `sizes` array(s), this message still gets logged

File: src/sizeMapping.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    });
+    if(mapping.sizes) {
+      return mapping.sizes;
+    }
+    return adUnit.sizes;
+  }
+  const sizes = adUnit.sizeMapping.find(sizeMapping =>{
+    return width > sizeMapping.minWidth;
+  }).sizes;
+  utils.logMessage(`AdUnit : ${adUnit.code} resized based on device width to : ${adUnit.sizes}`);
```
Comment (matthewlane) [inline src/sizeMapping.js]: If a `sizeMapping` is configured, is that meant to replace the `sizes` property? If so this message will show width as `undefined`

File: test/spec/sizeMapping_spec.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+describe('sizeMapping', function() {
+
+  beforeEach(resetMockWindow);
+
+  it('mapSizes 1029 width', function() {
+    let stub = sinon.stub(sizeMapping, 'getScreenWidth').returns(1029);
+    let sizes = sizeMapping.mapSizes(validAdUnit);
+    console.log(sizeMapping.getScreenWidth());
```
Comment (matthewlane) [inline test/spec/sizeMapping_spec.js]: May want to remove this line

**Comment by mkendall07** [comment]
@matthewlane 
PR notes addressed.

File: src/sizeMapping.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ */
+import * as utils from './utils';
+let _win;
+
+function mapSizes(adUnit) {
+  if(!isSizeMappingValid(adUnit.sizeMapping)){
+    return adUnit.sizes;
+  }
+  const width = getScreenWidth();
+  console.log('internal screen width: ' + width);
```
Comment (matthewlane) [inline src/sizeMapping.js]: console.log snuck in here

**Pull request merged**